### PR TITLE
Node.js bindings: Add JS bindings for SPI

### PR DESCRIPTION
**Pull request by nagineni**
This patch contains the native and highlevel JavaScript bindings for SPI.

### CODE REVIEW

File: bindings/nodejs/src/functions/spi.cc [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    if (tx)
+        txBuffer = Nan::NewBuffer((char *)tx, status).ToLocalChecked();
+    else
+        txBuffer = Nan::Null();
+
+    if (rx)
+        rxBuffer = Nan::NewBuffer((char *)rx, status).ToLocalChecked();
+    else
+        rxBuffer = Nan::Null();
+
```
Comment (cabelitos) [inline bindings/nodejs/src/functions/spi.cc]: Hey, I think this still quite wrong. The last version was almost correct.
You should do something like.
```C
if (status >= 0) {
  txBuffer = Nan::NewBuffer((char *)tx, status).ToLocalChecked();
  rxBuffer = Nan::NewBuffer((char *)rx, status).ToLocalChecked();
} else {
  free(tx);
  free(tx);
  txBuffer = Nan::Null();
  rxBuffer = Nan::Null();
}
`

File: bindings/nodejs/src/functions/spi.cc [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    Nan::SetInternalFieldPointer(jsSpi, 0, 0);
+}
+
+static void sol_spi_transfer_cb(void *cb_data, struct sol_spi *spi,
+    const uint8_t *tx, uint8_t *rx, ssize_t status) {
+    Nan::HandleScope scope;
+    Nan::Callback *callback = (Nan::Callback *)cb_data;
+    Local<Value> txBuffer, rxBuffer;
+
+    if (tx)
```
Comment (cabelitos) [inline bindings/nodejs/src/functions/spi.cc]: This will never be null

File: bindings/nodejs/src/functions/spi.cc [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        txBuffer = Nan::NewBuffer((char *)tx, status).ToLocalChecked();
+    else
+        txBuffer = Nan::Null();
+
+    if (rx)
```
Comment (cabelitos) [inline bindings/nodejs/src/functions/spi.cc]: this will never be null

File: bindings/nodejs/src/functions/spi.cc (hunk 2 above)
Comment (nagineni) [inline bindings/nodejs/src/functions/spi.cc]: Ah! Got the point. We have null checks for tx & rx in bind_sol_spi_transfer.

**Pull request closed without merging**